mesonbuild/interpreter.py

```diff
@@ -2581,6 +2581,8 @@ def run_command_impl(self, node, args, kwargs, in_builddir=False):
                       ' and therefore cannot be used during configuration'
                 raise InterpreterException(msg.format(progname, cmd.description()))
             if not cmd.found():
+                if isinstance(cmd, dependencies.ExternalProgram):
+                    raise InterpreterException('Tried to use not-found external program')
```

Comment:
Nitpicking: I would the external program's name in the message: cmd.get_name().

Reply:
Right, I'd rather return the external program's name. But the `cmd.get_name()` will return `nonexistingprogram` as it's an instance of `NonExistingExternalProgram`.
This commit fixes the bad error message. Further patches could deal with the implementation itself to enable showing the non-found program's name in the message.

Reply:
More specifically, the `NonExistingExternalProgram` instance is being created in the `Interpreter::find_program_impl` method with an empty name argument, which defaults to `nonexistingprogram`. This could be changed in order to store the name of the target program, even if it hasn't been found. I wonder if there's any reason not to do so.

Reply:
That can be improved by returning `NonExistingExternalProgram(args[0])` in `func_find_program()` and `find_program_impl()`

Reply:
Or even `', '.join(args)` instead of `args[0]`

Reply:
Exactly, I agree. I'll submit another PR setting the proper name. I'd rather use the joined args array, as that is more correct than just the first program name.

Reply:
After chatting about this, I'm closing this PR as I think the proper solution would be storing the program's name in the `NonExistingExternalProgram` instances, and showing it in the current error message. @xclaesse thanks!